### ISSUE TITLE
4.7-5

### ISSUE THREAD

**Issue by wojtask** [wojtask/clrs4e-solutions#141]

**Comment by wojtask** [comment]
Part (b) has a nasty example, because it's impossible to find an exact value for p, it's 1<p<2. Then, the integral required in the Akra-Bazzi method is \int\frac{dx}{x^{p-1}\lg x} which is an exponential integral, equal to ln2 Ei((2-p)ln x), and the solution to the recurrence is T(n)=\Theta(x^p Ei((2-p)\ln x)).

My guess is that it was unintended, and I believe that slightly modifying a_i, b_i, or the function f(n) can result in an easier recurrence to solve (even without knowing the exact p). I'm going to contact the Authors about that.

**Comment by wojtask** [comment]
Still no response from the authors.

In the meantime I came to realize that it's not the exponential integral that's a real problem here, but the fact that the definite integral probably doesn't converge. The integral we need to calculate is $\int h(x)dx$, where
```math
h(x)=\frac{x^{1-p}}{\lg x}.
```
The definite integral can be viewed as an area between the function plot and the x axis, bounded by the integral bounds (here: 1 and n). Since $\lim_{n\to1^+}h(x)=+\infty$, that area is infinite.

See [result](https://www.wolframalpha.com/input?i2d=true&i=Integrate%5BDivide%5BPower%5Bx%2C-0.85674%5D%2CLog%5B2%2Cx%5D%5D%2C%7Bx%2C1%2Cn%7D%5D) from WolframAlpha.

I'm going to conclude that the recurrence cannot be solved with the Akra-Bazzi method. I will leave a note saying that the example is most likely invalid.

**Comment by wojtask** [comment]
I managed to prove the asymptotic tight bound, despite the difficulties mentioned above. I showed the lower bound using the Akra-Bazzi method as required, after bounding the integral, which overcame the fact it's divergent. Then, I managed to prove that the bound is tight, using the substitution method.

Nevertheless, I am going to update the solution, once the authors come back to me saying they have modified this example.
